components/ntp_background_images/browser/view_counter_service.cc

```diff
@@ -314,6 +323,11 @@ void ViewCounterService::BrandedWallpaperLogoClicked(
   ads_service_->TriggerNewTabPageAdEvent(
       wallpaper_id, creative_instance_id,
       ads::mojom::NewTabPageAdEventType::kClicked);
+
+  if (ntp_p3a_helper_ && !ads_service_->IsEnabled()) {
+    // Should only report to P3A if ads are disabled, as required by spec.
+    ntp_p3a_helper_->RecordClickAndMaybeLand(creative_instance_id);
+  }
```

Comment:
@aseren, this code should only be called when an NTP-SI has been clicked if a user has not joined, rewards/ads are disabled. Can you please review this change? Thanks

Reply:
This should work fine